proxy/src/http.rs

```diff
@@ -21,6 +24,23 @@ pub fn new_client() -> ClientWithMiddleware {
         .build()
 }
 
+pub fn new_client_with_timeout_and_retries(
+    default_timout: Duration,
+    total_retries: u32,
+) -> ClientWithMiddleware {
+    let timeout_client = reqwest::ClientBuilder::new()
+        .timeout(default_timout)
+        .build()
+        .expect("Failed to create http client with timeout");
+
+    let retry_policy = ExponentialBackoff::builder().build_with_max_retries(total_retries);
+
+    reqwest_middleware::ClientBuilder::new(timeout_client)
+        .with(reqwest_tracing::TracingMiddleware::default())
+        .with(RetryTransientMiddleware::new_with_policy(retry_policy))
```

Comment:
```suggestion
        // As per docs, "This middleware always errors when given requests with streaming bodies".
        // That's all right because we only use this client to send `serde_json::RawValue`, which
        // is not a stream.
        .with(RetryTransientMiddleware::new_with_policy(retry_policy))
```